src/VisualStudio/Core/Def/CallHierarchy/CallHierarchyCommandHandler.cs

```diff
@@ -74,7 +75,14 @@ public bool ExecuteCommand(ViewCallHierarchyCommandArgs args, CommandExecutionCo
 
                     if (mapping.Symbol != null)
                     {
-                        var node = _provider.CreateItemAsync(mapping.Symbol, mapping.Project, ImmutableArray<Location>.Empty, cancellationToken).WaitAndGetResult(cancellationToken);
+                        ICallHierarchyMemberItem node = null;
+
+                        // ICallHierarchyMemberItem needs to switch to the UI thread to get the INavigableLocation
+                        _threadingContext.JoinableTaskFactory.Run(async () =>
```

Comment:
💭 Ideally _all_ the asynchronous operations in this method would be invoked in this fashion (or have the contents of this method wrapped in an asynchronous method), but this should at least work assuming there are no other unknown context switches

Reply:
I'd prefer we do what Sam mentions. It will also make this code more consistent with the rest of the product. We could also just make this experience fully async, using a twd to let the user know we're doing work, rather than just hanging the UI thread.